### ISSUE TITLE
benchmark result tag key/value pairs: specification precision: store those with null/None value (and make those part of 'case')?

### ISSUE THREAD

**Issue by jgehrcke** [conbench/conbench#939]
I see that from the database we get such key/value pairs:
```
conbench-app-1  | {'Python_specific_tags_field_1': 'value-1', 'Python_specific_tags_field_2': 'Python-tags-value-2', 'compression': 'uncompressed1', 'cpu_count': None, 'dataset': 'fanniemae_2016Q4', 'field_5': 'value-5', 'field_6': 'value-6', 'id': 'aebc199a755d443a8c7ecb687c84927f', 'name': 'csv-write', 'streaming': 'streaming', 'tags_field_3': 'Python-tags-value-3', 'tags_field_4': None}
```
Note those with value `None`. Currently we store those, and they seem to contribute to the case permutation hash/index. We have often not emitted those however, when stringifying the case.

I think we need to sharpen the specification around key type (string) and value types of case tags. I think a flat str/str mapping can be sufficient, can it?

**Comment by alistaire47** [comment]
There's an issue with voltrondata-labs/benchmarks that it's passing `"cpu_count": None` when really that means "use everything", i.e. `machine_info.cpu_core_count`. For R benchmarks, switching to running directly through arrowbench would populate this, but we'll need to go insert the values if we want to unbreak histories.

**Comment by alistaire47** [comment]
More generally, I'm of the opinion it would be good to fix the `None` issue and require values be populated. I'm less convinced it's worth it to require all values are strings; `"scale_factor": "10"` or `"memory_mapped": "False"` strike me as worse than their typed versions.

**Comment by jgehrcke** [comment]
highly related: #940

**Comment by jgehrcke** [comment]
> There's an issue with voltrondata-labs/benchmarks that it's passing "cpu_count": None when really that means "use everything"

That's a good one! "all-available-cores" might be a nice string representation for this; and this is maybe even a surprising example for when adding type constraints might have helped sending a more descriptive value.

> "scale_factor": "10" or "memory_mapped": "False" strike me as worse than their typed versions.

For connecting the dots: this is mainly discussed in issue #940 and also on PR #948.

**Comment by jgehrcke** [comment]
I think we can close this one here as 'done' for now after having landed #948. Let's go back to this discussion here / open more specific tickets when the need arises.

It might be interesting to explore removing all keys with `None` values in existing databases, but I am more forward-looking than backwards-looking today. :)
